src/backend/renderer/pixman/mod.rs

```diff
@@ -0,0 +1,1183 @@
+//! Implementation of the rendering traits using pixman
+use std::{
+    collections::HashSet,
+    rc::Rc,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+use drm_fourcc::{DrmFormat, DrmFourcc, DrmModifier};
+use pixman::{Filter, FormatCode, Image, Operation, Repeat};
+use tracing::warn;
+
+use crate::{
+    backend::allocator::{
+        dmabuf::{DmaBufSyncFlags, Dmabuf, DmabufMapping, WeakDmabuf},
+        format::has_alpha,
+        Buffer,
+    },
+    utils::{Buffer as BufferCoords, Physical, Rectangle, Scale, Size, Transform},
+};
+
+#[cfg(feature = "wayland_frontend")]
+use crate::{
+    backend::renderer::{ImportDmaWl, ImportMemWl},
+    wayland::{compositor::SurfaceData, shm},
+};
+#[cfg(feature = "wayland_frontend")]
+use wayland_server::{protocol::wl_buffer, Resource, Weak};
+
+#[cfg(all(
+    feature = "wayland_frontend",
+    feature = "backend_egl",
+    feature = "use_system_lib"
+))]
+use super::ImportEgl;
+use super::{
+    sync::SyncPoint, Bind, DebugFlags, ExportMem, Frame, ImportDma, ImportMem, Offscreen, Renderer, Texture,
+    TextureFilter, TextureMapping, Unbind,
+};
+
+mod error;
+
+pub use error::*;
+
+lazy_static::lazy_static! {
+    static ref SUPPORTED_FORMATS: Vec<drm_fourcc::DrmFourcc> = {
+        vec![
+            #[cfg(target_endian = "little")]
+            DrmFourcc::Rgb565,
+            DrmFourcc::Xrgb8888,
+            DrmFourcc::Argb8888,
+            DrmFourcc::Xbgr8888,
+            DrmFourcc::Abgr8888,
+            DrmFourcc::Rgbx8888,
+            DrmFourcc::Rgba8888,
+            DrmFourcc::Bgrx8888,
+            DrmFourcc::Bgra8888,
+            #[cfg(target_endian = "little")]
+            DrmFourcc::Xrgb2101010,
+            #[cfg(target_endian = "little")]
+            DrmFourcc::Argb2101010,
+            #[cfg(target_endian = "little")]
+            DrmFourcc::Xbgr2101010,
+            #[cfg(target_endian = "little")]
+            DrmFourcc::Abgr2101010,
+        ]
+    };
+}
+
+#[derive(Debug)]
+enum PixmanTarget {
+    Image { dmabuf: Dmabuf, image: PixmanImage },
+    RenderBuffer(PixmanRenderBuffer),
+}
+
+impl PixmanTarget {
+    fn image(&self) -> &pixman::Image<'static, 'static> {
+        match self {
+            PixmanTarget::Image { image, .. } => &image.0.image,
+            PixmanTarget::RenderBuffer(render_buffer) => &render_buffer.0,
+        }
+    }
+}
+
+/// Offscreen render buffer
+#[derive(Debug)]
+pub struct PixmanRenderBuffer(pixman::Image<'static, 'static>);
```

Comment:
Can't we make this take a proper lifetime?
I find it kinda annoying, that for example you couldn't use this renderer in combination with [`softbuffer`](https://docs.rs/softbuffer), as there is no way to `bind` the slice it provides you to render into.

Reply:
That would require changing the `Bind` trait I guess, not sure if this should be part of this PR?
Probably also requires a bigger refactor of the rendering traits in general. The renderer currently stores
the bind target in the renderer. So if we introduce some lifetime here we would also need to introduce that on the renderer.

Reply:
created a separate issue for this